packages/glob/src/internal-path-helper.ts

```diff
@@ -0,0 +1,144 @@
+import * as assert from 'assert'
+import * as path from 'path'
+
+const IS_WINDOWS = process.platform === 'win32'
+
+/**
+ * Roots the path if not already rooted
+ */
+export function ensureRooted(root: string, p: string): string {
+  assert(root, `ensureRooted parameter 'root' cannot be empty`)
+  assert(p, `ensureRooted parameter 'p' cannot be empty`)
+
+  // Already rooted
+  if (isRooted(p)) {
+    return p
+  }
+
+  // On Windows, check for root like C:
+  if (IS_WINDOWS && root.match(/^[A-Z]:$/i)) {
+    return root + p
```

Comment:
Should these path creations be using `path.join`? https://nodejs.org/api/path.html#path_path_join_paths

If `root` does not end in a trailing separator (`C:`), won't you hit issues if `path` doesn't begin with a separator?

Reply:
`path.join('C:', 'foo')` returns `C:\foo` which is wrong

https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats